src/osd/OSD.cc

```diff
@@ -1688,26 +1688,6 @@ void OSDService::queue_for_peering(PG *pg)
 
 void OSDService::queue_for_snap_trim(PG *pg)
 {
-  class PGSnapTrim : public PGOpQueueable {
```

Comment:
@myoungwon could you elaborate a little bit why moving `PGSnapTrim` out of the `OSDService::queue_for_snap_trim()` can help with the generalization of `PGSnapTrim`?

Reply:
@tchaikov This is because create_snaptrim(), create_srub(), create_recovery() in src/test/osd/TestMClockClientQueue.cc  create their instances directly without using queue_for_*(). I will refine it.